hpy/devel/include/hpy/hpydef.h

```diff
@@ -114,12 +114,12 @@ typedef struct {
    more detailed explanation in the comments around HPyFunc_DECLARE in
    hpyfunc.h
 */
-#define HPyDef_SLOT(SYM, IMPL, SLOT)                            \
+#define HPyDef_SLOT_IMPL(SYM, IMPL, SLOT)                            \
```

Comment:
The philosophy we followed so far was that we only provide a macro for the very common case.
I'm not sure if we should keep the `HPyDef_*_IMPL` macros since you can always manually fill your `HPyDef` struct.

Reply:
My own view is that manually filling the HPyDef struct is quite a big jump in complexity for users. One has to suddenly understand all the details of the struct.

Reply:
I agree since I cannot fill an `HPyDef` from the top of my head myself.
The question is more: where should we make the cut. I remember that I once asked if we should have some variant for `HPyDef_METH` that covers the _ujson_ case (same C function; two Python methods) and we decided to not introduce a macro for that.
Anyway, I'm not really objected to keep the `HPyDev_*_IMPL` macros. I would also agree to add more. We just need to start writing docs about that.

Reply:
`HPyDev_*_IMPL` can be good if you want to implement a slot with one of HPy's generic functions. I saw one in a test case